internal/db/schema/migrations/oss/postgres/88/01_remove_session_connection_state.up.sql

```diff
@@ -0,0 +1,224 @@
+                connection_id,
+                session_id,
+                host_key,
+                user_key,
+                credential_group_key,
+                connection_authorized_date_key,
+                connection_authorized_time_key,
+                connection_authorized_time,
+                client_tcp_address,
+                client_tcp_port_number,
+                endpoint_tcp_address,
+                endpoint_tcp_port_number,
+                bytes_up,
+                bytes_down
+    )
+    select new.public_id,
+           new.session_id,
+           session_dimension.host_dim_key,
+           session_dimension.user_dim_key,
+           session_dimension.credential_group_dim_key,
+           authorized_timestamp.date_dim_key,
+           authorized_timestamp.time_dim_key,
+           authorized_timestamp.ts,
+           new.client_tcp_address,
+           new.client_tcp_port,
+           new.endpoint_tcp_address,
+           new.endpoint_tcp_port,
+           new.bytes_up,
+           new.bytes_down
+      from authorized_timestamp,
+           session_dimension
+ returning * into strict new_row;
+    return null;
+  end;
+  $$ language plpgsql;
+
+  create trigger wh_insert_session_connection after insert on session_connection
+    for each row execute function wh_insert_session_connection();
+
+  create function wh_insert_session_connection_state() returns trigger
+  as $$
+    declare
+         state text;
+      date_col text;
+      time_col text;
+        ts_col text;
+             q text;
+      connection_row wh_session_connection_accumulating_fact%rowtype;
+    begin
+      if new.connected_time_range is null then
+        -- Indicates authorized connection. The update statement in this
+        -- trigger will fail for the authorized state because the row for the
+        -- session connection has not yet been inserted into the
+        -- wh_session_connection_accumulating_fact table.
+        return null;
+      end if;
+
+      if upper(new.connected_time_range) = 'infinity'::timestamptz then
+            update wh_session_connection_accumulating_fact
+               set (connection_connected_date_key,
+                    connection_connected_time_key,
+                    connection_connected_time) = (
+             select wh_date_key(new.update_time),
+                    wh_time_key(new.update_time),
+                    new.update_time::timestamptz)
+              where connection_id = new.public_id;
+        --  returning *;
+      else
+          update wh_session_connection_accumulating_fact
+                set (connection_closed_date_key,
+                     connection_closed_time_key,
+                     connection_closed_time) = (
+              select wh_date_key(new.update_time),
+                     wh_time_key(new.update_time),
+                     new.update_time::timestamptz)
+               where connection_id = new.public_id;
```

Comment:
Same thing here:

```suggestion
          update wh_session_connection_accumulating_fact
             set (connection_closed_date_key,
                  connection_closed_time_key,
                  connection_closed_time) = (select wh_date_key(new.update_time),
                                                    wh_time_key(new.update_time),
                                                    new.update_time::timestamptz)
           where connection_id = new.public_id;
```

Reply:
Is it possible we could have an autoformatter for sql? I'm going to admit sometimes I think I'm doing river and then I'm not sure why something isn't, and it can get frustrating.

Reply:
> Is it possible we could have an autoformatter for sql

We don't have anything currently :(.

> sometimes I think I'm doing river and then I'm not sure why something isn't

In this case, you want to separate the sub-query into its own river so it is more obvious where the sub-query begins and ends. See [this example](https://github.com/hashicorp/boundary/blob/ad4957a3c7d459be1fbbd8b9d24ff9f6b666ede1/sql-style-guide.md?plain=1#L173-L216).